artifactory/commands/python/poetry.go

```diff
@@ -138,37 +136,29 @@ func (pc *PoetryCommand) SetCommandName(commandName string) *PoetryCommand {
 }
 
 func (pc *PoetryCommand) SetPypiRepoUrlWithCredentials() error {
-	rtUrl, err := url.Parse(pc.serverDetails.GetArtifactoryUrl())
+	rtUrl, username, password, err := GetPypiRepoUrlWithCredentials(pc.serverDetails, pc.repository)
 	if err != nil {
-		return errorutils.CheckError(err)
-	}
-
-	username := pc.serverDetails.GetUser()
-	password := pc.serverDetails.GetPassword()
-
-	// Get credentials from access-token if exists.
-	if pc.serverDetails.GetAccessToken() != "" {
-		if username == "" {
-			username = auth.ExtractUsernameFromAccessToken(pc.serverDetails.GetAccessToken())
-		}
-		password = pc.serverDetails.GetAccessToken()
+		return err
 	}
-	rtUrl.Path += "api/pypi/" + pc.repository + "/simple"
 	if password != "" {
-		return pc.configPoetryRepo(rtUrl.Scheme+"://"+rtUrl.Host+rtUrl.Path, username, password)
+		return ConfigPoetryRepo(
+			rtUrl.Scheme+"://"+rtUrl.Host+rtUrl.Path,
+			username,
+			password,
+			pc.poetryConfigRepoName)
 	}
 	return nil
 }
 
-func (pc *PoetryCommand) configPoetryRepo(url, username, password string) error {
+func ConfigPoetryRepo(url, username, password, configRepoName string) error {
```

Comment:
This function has been made public taken out of the `PoetryCommand` struct, so that it can be accessed from an Xray command.
Having Xray commands using APIs that are included inside files of Artifactory commands isn't a good approach in my opinion. This can lead to complicated code flows over time. If we have APIs that need to be used by multiple commands, especially if these are Artifactory and Xray commands, we should move these APIs into util files, that are outside of the `commands` package. We may need to create `pythonutils.go` file for this.
I think this comment applies to other changes made as part of this PR as well, so let's do the same for all of them.

Reply:
Changes were made, reassigning you for re-review.